lib/components/viewers/route-viewer.js

```diff
@@ -51,47 +69,91 @@ class RouteViewer extends Component {
           <div className='header-text'>
             {languageConfig.routeViewer || 'Route Viewer'}
           </div>
+          <div className=''>
+            {languageConfig.routeViewerDetails}
+          </div>
           <div style={{ clear: 'both' }} />
         </div>
 
         <div className='route-viewer-body'>
-          {routes && Object.values(routes).sort(routeComparator).map(route =>
-            <RouteRow key={route.id} route={route} {...this.props} />)}
+          {agencySortedRoutes
+            .map(route => {
+              // Find operator based on agency_id (extracted from OTP route ID).
+              const operator = operatorForRoute(operators, route) || {}
+              return (
+                <RouteRow
+                  key={route.id}
+                  operator={operator}
+                  route={route}
+                  {...this.props} />
+              )
+            })
+          }
         </div>
       </div>
     )
   }
 }
 
 class RouteRow extends Component {
```

Comment:
With the React Developer tools, it can be seen that clicking on a new route results in a very sluggish css transition. This is happening largerly because each RouteRow is freshly rendered whenever another RouteRow changes. All of these renders are not needed since clicking on one route shouldn't require a rerender of another RouteRow 20 spots above. A simple fix for this is to change the RouteRow to extend PureComponent and make sure the input props change only when absolutely needed.

See this article about [how PureComponents can help with performance](https://blog.logrocket.com/make-react-fast-again-part-2-why-did-you-update-dd1faf79399f/).

Reply:
I'll approve this PR and make a separate PR to dev in the interest of moving things along.